SocketRocket/SRWebSocket.m

```diff
@@ -373,6 +373,25 @@ - (void)_HTTPHeadersDidFinish;
 {
     NSInteger responseCode = CFHTTPMessageGetResponseStatusCode(_receivedHTTPHeaders);
     if (responseCode >= 400) {
+        
+        if (responseCode == 401)
```

Comment:
You should also check for a "WWW-Authenticate" HTTP header in the response identifying the "Bearer" scheme before relying on the authentication delegate to resolve. There are other reasons for 401 responses